internal/db/rdb/mysql/mysql.go

```diff
@@ -122,7 +125,7 @@ func (m *mySQLClient) Open(ctx context.Context) error {
 	conn.SetMaxIdleConns(m.maxIdleConns)
 	conn.SetMaxOpenConns(m.maxOpenConns)
 
-	m.session = conn.NewSession(nil)
+	m.session = dbr.NewSession(*conn, nil)
```

Comment:
can we put `NewSession()` to `m.dbr`?

Reply:
Did you mean `m.session = m.dbr.NewSession(*conn, nil)` ?

Reply:
@kevindiu

Reply:
@vankichi 🤔🤔 maybe current design is better than `m.dbr.NewSession()` 🤔

Reply:
but can we mock `dbr.Connection`?

Reply:
if we can do this, we can follow the origin code instead of creating our custom function

Reply:
I am not sure if we can do it:
change the return type of https://github.com/vdaas/vald/pull/659/files#diff-c05bae65105c786c32926dfcd653e9a4R23 from `*Connection` to `Connection` and define our own mock `Connection` interface

Reply:
I think it depends on use case and if we can create it individually

Reply:
@kevindiu 
during considering, in this situation, we do not have to this.
`NewSession()` use the Connection which will be created by `m.dbr.New()`, in other word, `NewSession` is not individual.

Reply:
@vankichi 
I think both fine. What I thought that following the original API `conn.NewSession()`. 
If we create the mock of the `Connection`,
https://github.com/vdaas/vald/pull/659/files#diff-c05bae65105c786c32926dfcd653e9a4R29
and change the 
https://github.com/vdaas/vald/pull/659/files#diff-c05bae65105c786c32926dfcd653e9a4R39 
to `func (*db) Open(driver, dsn string, log EventReceiver) (Connection, error) {`, 
we can mock everything.

If you think your implementation is good enough, I think it is ok